rust/lance/src/io/commit/conflict_resolver.rs

```diff
@@ -0,0 +1,722 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use std::{
+    borrow::Cow,
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
+
+use futures::{StreamExt, TryStreamExt};
+use lance_core::{
+    utils::{deletion::DeletionVector, mask::RowIdTreeMap},
+    Result,
+};
+use lance_table::{
+    format::Fragment,
+    io::deletion::{deletion_file_path, read_deletion_file_cached, write_deletion_file},
+};
+use snafu::{location, Location};
+
+use crate::{
+    dataset::transaction::{ConflictResult, Operation, Transaction},
+    Dataset,
+};
+
+pub struct TransactionRebase<'a> {
+    transaction: Transaction,
+    /// Relevant fragments as they were at the read version of the transaction.
+    /// Has original fragment, plus a bool indicating whether a rewrite is needed.
+    initial_fragments: HashMap<u64, (Fragment, bool)>,
+    affected_rows: Option<&'a RowIdTreeMap>,
+}
+
+impl<'a> TransactionRebase<'a> {
+    pub async fn try_new(
+        dataset: &Dataset,
+        transaction: Transaction,
+        affected_rows: Option<&'a RowIdTreeMap>,
+    ) -> Result<Self> {
+        // We might not need to check for row-level conflicts anyways.
+        if !matches!(&transaction.operation,
+            Operation::Update { updated_fragments, .. } |
+            Operation::Delete { updated_fragments, .. }
+            if !updated_fragments.is_empty() && affected_rows.is_some(),
+        ) {
+            return Ok(Self {
+                transaction,
+                initial_fragments: HashMap::new(),
+                affected_rows: None,
+            });
+        }
+
+        log::info!(
+            "Manifest version: {}, transaction version: {}",
+            dataset.manifest.version,
+            transaction.read_version
+        );
```

Comment:
Do we intend to keep this log message?  If so, can we add context?  Something like `Rebasing tx at version {} on top of dataset at version {}"

Reply:
oh forgot to remove it, it was for debugging